src/org/labkey/test/util/core/webdav/WebDavUploadHelper.java

```diff
@@ -133,6 +133,36 @@ public void uploadFile(@NotNull File file)
         uploadFile(file, "");
     }
 
+    public void deleteFile(@NotNull String file, boolean shouldFail)
+    {
+        try
+        {
+            _sardine.delete(file);
+        }
+        catch (IOException e)
+        {
+            if (shouldFail)
+              ;
+            else
+                throw new RuntimeException(e);
+        }
```

Comment:
This is a bit too wide of a net for an expected error. An IOException could be all sorts of things. I would at least make sure the correct exception type was thrown; then return it so the test could verify the results.
e.g.
```
public SardineException deleteExpectingError(String file)
{
    try
    {
        _sardine.delete(_urlFactory.getPath(file));
        throw new RuntimeException("Expected error did not occur");
    }
    catch (SardineException expected)
    {
        return expected;
    }
    catch (IOException e)
    {
        throw new RuntimeException(e);
    }
}
```
`SardineException` will include a status code which would be good for verifying in this test.